lib/Agrammon/DB/User.pm6

```diff
@@ -15,13 +15,21 @@ class X::Agrammon::DB::User::NoUsername is Exception {
     }
 }
 
+class X::Agrammon::DB::User::CreateFailed is Exception {
+    has Str $.user-name is required;
+
+    method message {
+        "Couldn't create user $.username";
```

Comment:
Should be `$.user-name` (but you don't really need it to be a virtual call at all, so you can access the underlying attribute directly as `$!user-name`, and those typos are caught at compile time).

Reply:
Fixed.